dune-project

```diff
@@ -1,3 +1,24 @@
-(lang dune 1.7)
+(lang dune 3.17)
```

Comment:
This is a big jump and makes it much less likely that people will be able to use notty if they require an older dune (which is very rare). I would pick `3.0` for the time being. If we need newer versions we can bump.